pkg/rotator/rotator.go

```diff
@@ -440,47 +488,46 @@ func (m *mapper) Map(object handler.MapObject) []reconcile.Request {
 }
 
 // add adds a new Controller to mgr with r as the reconcile.Reconciler
-func addController(mgr manager.Manager, r reconcile.Reconciler, secretKey, vwhKey types.NamespacedName) error {
-	// Create a new controller
-	c, err := controller.New(("validating-webhook-controller"), mgr, controller.Options{Reconciler: r})
-	if err != nil {
-		return err
-	}
-
-	// Watch for changes to the provided ValidatingWebhookConfiguration
-	s := &corev1.Secret{}
-	if err := c.Watch(&source.Kind{Type: s}, &handler.EnqueueRequestForObject{}); err != nil {
-		return err
-	}
-
+func addController(mgr manager.Manager, r *ReconcileWH) error {
 	vwh := &unstructured.Unstructured{}
 	vwh.SetGroupVersionKind(vwhGVK)
-	mapper := &handler.EnqueueRequestsFromMapFunc{ToRequests: &mapper{
-		secretKey: secretKey,
-		vwhKey:    vwhKey,
-	}}
-	if err := c.Watch(&source.Kind{Type: vwh}, mapper); err != nil {
+	crd := &unstructured.Unstructured{}
+	crd.SetGroupVersionKind(crdGVK)
+	// Create a new controller
+	err := ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Secret{}).
+			Watches(&source.Kind{Type: vwh}, &handler.EnqueueRequestsFromMapFunc{ToRequests: &mapper{
+				secretKey: r.secretKey,
+				vwhKey:    r.vwhKey,
+			}}).
+			Watches(&source.Kind{Type: crd}, &handler.EnqueueRequestsFromMapFunc{ToRequests: &crdMapper{
```

Comment:
What do we do if there are no CRD conversion webhooks we want to watch?

Reply:
The `Map` function only returns the `secretKey` if for the matching CRDs. If the `CRDNames []string` is empty, the reconciler will not be triggered for any CRDs.

To inject CA only to VWH just like before, users just don't set `CRDNames` field in `CertRotator`.